lisa/git.py

```diff
@@ -42,7 +42,7 @@ def find_shortest_symref(repo_path, sha1):
         cwd=repo_path, shell=True)
     for line in branches.splitlines():
         try:
-            sha1_out, name = line.strip().split()
+            sha1_out, name = line.strip().decode('utf8').split()
```

Comment:
IMO that should be fixed by passing `universal_newlines=True` to the above `check_output()` call. See https://github.com/ARM-software/devlib/commit/78de479a43466d33050e8573fb264b567dd2d81b for an example :)

Reply:
Yes, do you think that will be more run-time efficient?

From the documentation I see that:

_"Changed in version 3.7: Added the text parameter, as a more understandable alias of universal_newlines. Added the capture_output parameter."_

and

_"The universal_newlines argument is equivalent to text and is provided for backwards compatibility. By default, file objects are opened in binary mode."_

Thus, I'm wondering if by using `universal_newlines` we will not be exposed to that param being deprecated in the future and thus being forced to change it again to `text`.
But, we will probably have to to that anyway for the usage we already have... thus, if there is a strong preference, I'll update the patch.

Let's wait for @douglas-raillard-arm before.

Reply:
> Yes, do you think that will be more run-time efficient?

That won't make a difference, it just allows getting the string directly instead of having to call `decode()` all over the place (I personally don't mind one or the other, as long as people know what they are doing).

Regarding `text`, we have to support Python >= 3.5 so we can't use that more intuitive name :)

> Thus, I'm wondering if by using `universal_newlines` we will not be exposed to that param being deprecated in the future

I've never seen any parameter like that removed, that would cause huge breakage (with strictly no benefit here). Some functions may start raising different kind of exceptions, but that is about the worst I've seen and they only do that in modules that are not commonly used (like `importlib`)